src/reducers/algorithmsReducer.js

```diff
@@ -30,7 +32,23 @@ export default (state = INITIAL_STATE, { type, payload }) => {
         updateActivityList(payload),
       );
     case GET_ALGORITHMS_SUCCESS:
-      return state.setIn(['algorithms'], List(payload));
+      return state
```

Comment:
Why not have this separation directly from the back-end in the listener? The payload could look like `{ algorithms: [...], utils: [...] }`